### DISCUSSION TITLE
Welcome to localGPT Discussions!

### DISCUSSION THREAD

**Discussion by PromtEngineer** [PromtEngineer/localGPT#8] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Allaye** [comment]
@PromtEngineer Nice work you have over here, what are your long term plans for the project? i think i can have a PR for you, if you think it is something you want to work on long-time.

**Reply by PromtEngineer** [reply]
@Allaye thank you, I do have long terms plans for this project. I would like this to be an open-source information retrieval system. Will be adding a lot more features in the coming days/weeks. Would love for you to contribute to the project.

**Reply by Allaye** [reply]
@PromtEngineer this make sense, i will create some few PR then we can move from there.

**Reply by AlexsandroSantosdaSilva87** [reply]
okay week project good thanks

**Reply by brainlessguille** [reply]
can localgpt be implemented to to run one model that will select the appropriate model base on user input. example the user ask a question about gaming coding, then localgpt will select all the appropriated models to generate code and animated graphics exetera...  im currently working on this but. is challenging because i can only work on it during my time off from work and  I'm not a programmer and  I'm in the process of learning coding, ,this update of localgpt i haven't able to get it running yet im having problem with one of the packages

**Comment by Hubertine** [comment]
Je voudrais être aidée pour importer mon code.

**Comment by middhaGH** [comment]
I am getting below error 
Split into 60 chunks of text
Traceback (most recent call last):
  File "B:\Software\Anaconda\lib\site-packages\langchain\embeddings\huggingface.py", line 125, in __init__
    from InstructorEmbedding import INSTRUCTOR
ModuleNotFoundError: No module named 'InstructorEmbedding'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "D:\Coding\ChatGPT API Tutorial\localGPT\ingest.py", line 78, in <module>
    main()
  File "B:\Software\Anaconda\lib\site-packages\click\core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "B:\Software\Anaconda\lib\site-packages\click\core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "B:\Software\Anaconda\lib\site-packages\click\core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "B:\Software\Anaconda\lib\site-packages\click\core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "D:\Coding\ChatGPT API Tutorial\localGPT\ingest.py", line 69, in main
    embeddings = HuggingFaceInstructEmbeddings(model_name="hkunlp/instructor-xl",
  File "B:\Software\Anaconda\lib\site-packages\langchain\embeddings\huggingface.py", line 131, in __init__
    raise ValueError("Dependencies for InstructorEmbedding not found.") from e
ValueError: Dependencies for InstructorEmbedding not found.

**Reply by PromtEngineer** [reply]
@middhaGH make sure you install the required dependencies using pip install -r requirements.txt . That seems to be the issue.

**Reply by middhaGH** [reply]
ignore it fixed the issues 
here is the new issue after I ran ingest.py 

AssertionError: Torch not compiled with CUDA enabled

I went in an ran any runlocalGPT.PY way 
RuntimeError: [enforce fail at ..\c10\core\impl\alloc_cpu.cpp:72] data. DefaultCPUAllocator: not enough memory: you tried to        
allocate 180355072 bytes.

How much does it need  any way , I had 16G and it ran out

**Comment by jmerc77** [comment]
hi. I like pretty much anything that involve some kind of integrated circuit 😁. i even once wrote some asm for pic16f62x series chips for some of my small projects! I've programed in asm, a little of c++ (mainly with Arduino boards), quite a bit of vb.net, some python, the occasional web development (even JavaScript), and openSCAD designs for my 3d printer (also used G code a bit). heck, I've hand soldered a lot of circuits over the years. anyway...

I have a suggestion for the training data etc: Please do not make it politically biased (or biased in any topic ideally!) maybe opinionated prompts should show all angles of the topic in the response and state first-line that it is a matter of opinion. let them decide what to believe.

PS: I started learning the tech at an early age. some basic HTML during elementary and got my first robotics set when I was 10 (from Lego). same went for math. learned some algebra before taught in class, and some trig never taught in the schools I went too (not even my 2 yrs in college because I pretty much aced that on the pre exam).

**Reply by jmerc77** [reply]
adding the middle paragraph to ideas soon. seams like a better fit.

**Comment by ChettakattuA** [comment]
Hi,
I managed to make it run both terminal way and using API but the result is taking forever to load. I use GPU and just have 1 document on source dir but still it doesnt seem to run?

**Comment by Mathwhizdude** [comment]
On Jun 26, 2023, at 3:55 AM, ChettakattuA ***@***.***> wrote:﻿
Hi,
I managed to make it run both terminal way and using API but the result is taking forever to load. I use GPU and just have 1 document on source dir but still it doesnt seem to run?

Ok. You may need to scrape the data before—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

**Reply by ChettakattuA** [reply]
What you mean by scrape the data. the document is a simple pdf document with only text and no images

**Comment by CodeAKrome** [comment]
Taught myself 6502 assembly on an Apple II. Did first Linux install 2 years after it came out. I'm currently interested in NLP using Apache Spark. NER , Sentiment , Fake news / bias detection. I'm working on an RSS feed parser with NER and NERD . I got pyspark NER run to work last night, so I plan on making a repo with a CLI. I was looking at ingest.py and thinking mebbe it could be spark-ified as well. Was thinkin mebbe add RSS to UI.

Great work,
Hackers of the world unite!

**Comment by afcarzero1** [comment]
Have you thought about adding support to make the bot be able to describe the contents of a large repository? So that new developers on a team can easily ask questions about the code itself?

**Comment by colmduff** [comment]
Just an idea, but, I would love to see education embraced. 

I have mostly used chatgpt to assist in error interpretation, inspiration for the solving of specific code problems, and getting my own code for potential efficiency gains. It has really helped in terms of learning. It's like having a python expert always available who never gets tired of your constant questions. 

It would be great to see development of a tool that really tries to advance the educational potential. I have seen CS50 have attempted something like this, in terms of teaching, it would allow educators to build in guard rails to mitigate risks, but allow students to really take advantage of the potential.

**Comment by Juankydestroyer** [comment]
> # ~~````
**_[]()![20230819_013055.jpg](https://github.com/PromtEngineer/localGPT/assets/97413556/4c531eb9-85f6-4812-a5c4-71bbb2a4a103)

_**
````~~
